File: Lista_8/lista8_ex5.c
/* Exercício 5:
Faça uma função recursiva em C que calcula a média dos elementos de um vetor.
*/
#include <stdio.h>
#include <stdlib.h>

double medRecursivo(double vet[], int n, int tam);

int main () {
int tam=5;
double vet[] = {2, 3.5, 10, 5.2, 4.7};

printf("Elemento media: %.2lf\n", medRecursivo(vet, tam, tam));

return 0;
}

double medRecursivo(double vet[], int n, int tam) {
    double aux;
    if(n == 1) {
        return vet[0];
    } else {
        aux = medRecursivo(vet, n-1, tam);///Variavel somadora;
        if(n == tam) {
            return (vet[n-1] + aux)/tam;
        } else {
            return vet[n-1] + aux;
        }
    }
}

File: Lista_8/Lista8_ex5.py
def media_recursivo(vetor, n, tam):
    if n == 1:
        return vetor[0]
    else:
        aux = media_recursivo(vetor, n-1, tam)
        if n == tam:
            return (vetor[n-1] + aux) / tam
        else:
            return vetor[n-1] + aux

def main():
	vetor = [2, 3.5, 10, 5.2, 4.7]
	
	print("Media dos elementos: %.2f\n" % media_recursivo(vetor, len(vetor), len(vetor)))

if __name__ == "__main__":
	main()
